assets/src/edit-story/components/form/numeric.js

```diff
@@ -79,6 +79,49 @@ const Suffix = styled.span`
     `}
 `;
 
+const DisplayMode = {
+  Empty: 0,
+  Value: 1,
+};
+
+function isIterimValue(value, isFloat, canBeNegative) {
```

Comment:
Would be great to split out parts of the logic (e.g. the helpers perhaps) to prevent the file from getting bloated.
Or, @barklund, is this something you would do anyway as part of #4104 ?